haas/ext/switches/_console.py

```diff
@@ -119,6 +119,16 @@ def revert_port(self, port):
         self.exit_if_prompt()
         self.console.expect(self.config_prompt)
 
+    def _save_check(self, switch_type):
+        """checks the config file to see if switch should save or not"""
+
+        save = True
+        switch_ext = 'haas.ext.switches.' + switch_type
+        if (cfg.has_option(switch_ext, 'save') and
+           not cfg.getboolean(switch_ext, 'save')):
+            save = False
+        return save
```

Comment:
This function could be more concise; you can skip the assignment statements and just write:

```python
return (not cfg.has_option(switch_ext, 'save') or
        cfg.getboolean(switch_ext, 'save'))